src/libcore/iter.rs

```diff
@@ -25,23 +25,22 @@
 //! the `FromIterator` trait for creating a container from an iterator, and much
 //! more.
 //!
-//! ## Rust's `for` loop
+//! # Rust's `for` loop
 //!
-//! The special syntax used by rust's `for` loop is based around the `Iterator`
-//! trait defined in this module. For loops can be viewed as a syntactical expansion
+//! The special syntax used by rust's `for` loop is based around the `IntoIterator`
+//! trait defined in this module. `for` loops can be viewed as a syntactical expansion
 //! into a `loop`, for example, the `for` loop in this example is essentially
 //! translated to the `loop` below.
 //!
 //! ```
 //! let values = vec![1, 2, 3];
 //!
-//! // "Syntactical sugar" taking advantage of an iterator
-//! for &x in values.iter() {
+//! for x in values {
 //!     println!("{}", x);
 //! }
 //!
 //! // Rough translation of the iteration without a `for` iterator.
-//! let mut it = values.iter();
+//! let mut it = values.into_iter();
```

Comment:
Ah, looks like @alexcrichton fixed it in a fixup commit https://github.com/alexcrichton/rust/commit/72f59732d7974767650abfc58f8287212e5a1fba#diff-ced4ae040c5c8672d936a581401ef9ceR44